nx/lib/nx.ex

```diff
@@ -6149,8 +6149,33 @@ defmodule Nx do
     rtol = opts[:rtol]
     atol = opts[:atol]
 
-    # TODO: deal with non_finite entries by adding is_infinity and is_nan
-    all(less_equal(Nx.abs(subtract(a, b)), add(atol, multiply(rtol, Nx.abs(b)))))
+    a = to_tensor(a)
+    b = to_tensor(b)
+
+    nan_a = is_nan(a)
```

Comment:
Sounds good to me!

Reply:
Actually, I think we should do it like Numpy and have the `:equal_nan` be an option. This way there is no additional cost by default.

Reply:
@josevalim should we add `:equal_nan` to `equal`, `not_equal`,  `greater_equal` and `less_equal` as well?

Reply:
Sorry, I meant that Python has a `:equal_nan` option in all_close. That should be all we need here too, right?

Reply:
Yes, that's what I understood :)
But I think the same logic could apply on those operations as well, because they also check for equality involving nans

Reply:
I see. Let's go easy and add them if needed in the future.